src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -77,92 +78,122 @@ public static IEnumerable<VisualStudioInstance> GetInstances(string workingDirec
             }
         }
 
+        enum hostfxr_resolve_sdk2_flags_t
+        {
+            disallow_prerelease = 0x1,
+        };
+
+        enum hostfxr_resolve_sdk2_result_key_t
+        {
+            resolved_sdk_dir = 0,
+            global_json_path = 1,
+        };
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Auto)]
+        private delegate void hostfxr_resolve_sdk2_result_fn(
+                hostfxr_resolve_sdk2_result_key_t key,
+                string value);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Auto)]
+        private delegate void hostfxr_get_available_sdks_result_fn(
+                hostfxr_resolve_sdk2_result_key_t key,
+                [MarshalAs(UnmanagedType.LPArray, SizeParamIndex = 0)]
+                string[] value);
+
+        [DllImport("hostfxr", CharSet = CharSet.Auto, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern int hostfxr_resolve_sdk2(
+            string exe_dir,
+            string working_dir,
+            hostfxr_resolve_sdk2_flags_t flags,
+            hostfxr_resolve_sdk2_result_fn result);
+
+        [DllImport("hostfxr", CharSet = CharSet.Auto, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern int hostfxr_get_available_sdks(string exe_dir, hostfxr_get_available_sdks_result_fn result);
+
+        [DllImport("libc", ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern IntPtr realpath(string path, IntPtr buffer);
+
+        [DllImport("libc", ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern void free(IntPtr ptr);
+
+        private static string realpath(string path)
+        {
+            IntPtr ptr = realpath(path, IntPtr.Zero);
+            string result = Marshal.PtrToStringAuto(ptr);
+            free(ptr);
+            return result;
+        }
+
         private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
         {
-            const string DOTNET_CLI_UI_LANGUAGE = nameof(DOTNET_CLI_UI_LANGUAGE);
-            
-            Process process;
-            var lines = new List<string>();
-            try
+            string dotnetPath = null;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                process = new Process()
-                { 
-                    StartInfo = new ProcessStartInfo("dotnet", "--info")
+                foreach (string dir in Environment.GetEnvironmentVariable("PATH").Split(';'))
```

Comment:
```suggestion
                foreach (string dir in Environment.GetEnvironmentVariable("PATH").Split(Path.PathSeparator))
```

This provides semantic meaning and gets rid of semi-magic constants.